### DISCUSSION TITLE
2024 Cohort 2 - Group 16 Discussion

### DISCUSSION THREAD

**Discussion by sharon-digipres** [Digital-Preservation-Coalition/PythonStudyGroups#42] [General]
This is the discussion thread for those in **Group 16** of the July-December 2024 Cohort

**Comment by arhonemus** [comment]
Noting here how I used Jupyter in Visual Studio Code.

- I have the Jupyter extensions from Microsoft installed.
- I selected New File in VSCode.
- I selected Jupyter Notebook.
<img width="1840" alt="Screenshot 2024-08-21 at 2 54 52 PM" src="https://github.com/user-attachments/assets/fedf80c8-4ee8-46db-a274-be6003f26f95">

**Comment by jfcarrano** [comment]
# 2024-08-21 Python Study Group 16 meeting notes
## Administrative
- Too many group members are out on September 18 to make sense to meet. We will skip this meeting and our group will meet again on October 16
- Ima will take notes in October, Casey will lead
- Tracy will take notes in November
## Installing python/discussion of set up
Everyone has Python installed
Tracy has used Anaconda environment and VScode
- information IDEs: https://en.wikipedia.org/wiki/Integrated_development_environment
Al uses VScode, explored the jupyter notebook functionality
- any advantages?
	- unsure, was nice to have when going through exercise
- works some with pyenv
- uses mostly Macs, some on Windows/WSL, some linux
Casey installed anaconda, uses pycharm as the interpreter, uses homebrew, VScode, writing in terminal
Joe
- windows subsystem linux/notepad++
Ima
- uses VScode, has used notepad++, VScode

## Work on [API exercise](https://github.com/Digital-Preservation-Coalition/PythonStudyGroups/blob/main/How-Tos/Use%20Case%20-%20Working%20with%20apis.md)
- some issues getting the requests module
- discussed reasons to install pendulum in the script instead of on the terminal

**Comment by ioduok** [comment]
**Python Study Group 16 Notes [October 16, 2024]**

- Working through How-To’s – [Deleting Unwanted Files](https://github.com/Digital-Preservation-Coalition/PythonStudyGroups/blob/main/How-Tos/Use%20Case%20How%20To%20-%20Deleting%20Unwated%20FIles.md)
- Created dummy directories and files (or used existing directories and adding fake files for deletion)
- Instead of using the “thumbs.db” as indicated in exercise, using different file names (i.e. ‘fakefile.jpg’ or ‘fakefile.doc’) or extensions (i.e. “file1.txt”, “file2.txt”, etc.) to search for and delete
>- 	Use *if item.endswith('.txt'):* or *if item.startswith('fakefile'):* instead of *if item == 'Thumbs.db':*
- Use of ‘f’ string or format string to troubleshoot or debug a function before it executes
>- 	Example: print(f'FILES: {files}')
